workflow/controller/dag.go

```diff
@@ -515,10 +515,11 @@ func (woc *wfOperationCtx) buildLocalScopeFromTask(dagCtx *dagContext, task *wfv
 		}
 		prefix := fmt.Sprintf("tasks.%s", ancestor)
 		if ancestorNode.Type == wfv1.NodeTypeTaskGroup {
-			var ancestorNodes []wfv1.NodeStatus
-			for _, node := range woc.wf.Status.Nodes {
+			var childNodes = make([]wfv1.NodeStatus, len(ancestorNode.Children))
+			for i, nodeID := range ancestorNode.Children {
+				node := woc.wf.Status.Nodes[nodeID]
 				if node.BoundaryID == dagCtx.boundaryID && strings.HasPrefix(node.Name, ancestorNode.Name+"(") {
```

Comment:
Is this check now moot if we're accessing the nodes directly from the children field?

Reply:
thanks. i think this check is no need any more.